src/parser/shared/modules/resources/resourcetracker/ResourceTracker.ts

```diff
@@ -157,6 +159,15 @@ export default class ResourceTracker extends Analyzer {
    *  timestamp as a valid gain.
    */
   allowMultipleGainsInSameTimestamp = false;
+  /** Instead of calculating resource as rounded numbers, use decimal precision for granularity.
+   * This is needed for specs that have resource generation that isn't provided in whole numbers
+   * eg. Evoker's Essence */
+  useGranularity = false;
+  /** amount of decimals to use for granularity */
+  granularity = 2;
```

Comment:
the way this is usually handled is by multiplying by the number of places that you want. i.e. for a granularity of 2 as specified here, you multiply by 100. WCL does some of this internally

it looks like there are only 2 spots that `useGranularity` really comes into play, and one of those i'm not clear on (see below)

Reply:
Is this what you have you in mind?
`granularity = 100;`
```
const naturalGain = this.useGranularity
      ? Math.round(timePassedSeconds * lastUpdate.rate * granularity ) / granularity )
      : Math.round(timePassedSeconds * lastUpdate.rate); // whole number amount of resources pls

```

Reply:
rereading my comment, i'm not sure what im suggesting tbh D:

this is probably fine as-is